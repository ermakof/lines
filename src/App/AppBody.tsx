import React, { FC } from 'react';
import styled from '@emotion/styled';
import GameField from '@src/modules/GameField';
import { useSelector } from 'react-redux';
import HitParade from '@src/modules/HitParade';
import UserInfo from '@src/modules/UserInfo';
import { getAuth } from '@src/modules/Auth/authSaga';
import { IAuthState } from '@src/modules/Auth/model/IAuthState';

interface IRoot {
  active?: boolean;
}
const Root = styled.div<IRoot>`
  display: flex;
  height: 80vh;
  position: relative;
  overflow: auto;
  opacity: ${({ active }) => (active ? 1 : 0.3)};
`;

const AppBody: FC = () => {
  const auth: IAuthState = useSelector(getAuth);
  return (
    <Root role="bodyPanel" active={!!auth.userProfile}>
      <HitParade />
      <GameField />
      <UserInfo login={auth.userProfile?.login} />
    </Root>
  );
};

export default AppBody;
