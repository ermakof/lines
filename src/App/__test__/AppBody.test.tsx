import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { render, screen } from '@testing-library/react';

import AppBody from '@src/App/AppBody';
import appInitialState from '@src/App/initialState';
import authInitialState from '@src/modules/Auth/initialState';

jest.mock('react-redux', () => ({
  __esModule: true,
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

const mockSelectors = (selector: any, store: any) => {
  return selector(store);
};

describe('AppBody', () => {
  beforeEach(() => {
    (useDispatch as jest.Mock).mockImplementation(jest.fn());
  });

  it('Render <AppBody> without data', () => {
    const mockStore = {
      app: appInitialState,
      auth: authInitialState,
    };
    (useSelector as jest.Mock).mockImplementation((selector) => mockSelectors(selector, mockStore));
    const { asFragment } = render(<AppBody />);
    expect(asFragment()).toMatchSnapshot();
    const dataMessage = screen.getByRole(/bodyPanel/gi);
    expect(dataMessage).toBeInTheDocument();
  });

  it('Render <AppBody> with data', () => {
    const mockStore = {
      app: {
        gameLevel: '1',
        gameFieldPercentFilled: 10,
        gameFieldData: [1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
      },
      auth: authInitialState,
    };
    (useSelector as jest.Mock).mockImplementation((selector) => mockSelectors(selector, mockStore));
    const { asFragment } = render(<AppBody />);
    expect(asFragment()).toMatchSnapshot();
    const dataList = screen.getByRole(/bodyPanel/gi);
    expect(dataList).toBeInTheDocument();
    const dataItems = screen.getAllByRole(/cellContainer/gi);
    expect(dataItems.length).toBe(9);
  });
});
