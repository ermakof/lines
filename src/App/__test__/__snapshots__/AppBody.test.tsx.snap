// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBody Render <AppBody> with data 1`] = `
<DocumentFragment>
  <div
    class="css-cdwrp8"
    role="bodyPanel"
  >
    <div
      class="css-1n2h5h0"
      role="panelHitParade"
    >
      <div
        class="css-11wxzwa"
        role="rootGrid"
      >
        <p
          class="css-1890bmp"
        >
          Хит-парад
        </p>
      </div>
    </div>
    <div
      class="css-1n2h5h0"
    >
      <section
        class="css-rxkl80"
        role="cellsGrid"
      >
        <div
          class="css-1bqu5zx"
          role="cellContainer-0"
        >
          <div
            class="css-zzdbx3"
            role="cellContent-0"
          />
        </div>
        <div
          class="css-g263yg"
          role="cellContainer-1"
        >
          <div
            class="css-zzdbx3"
            role="cellContent-1"
          />
        </div>
        <div
          class="css-g263yg"
          role="cellContainer-2"
        />
        <div
          class="css-g263yg"
          role="cellContainer-3"
        />
        <div
          class="css-g263yg"
          role="cellContainer-4"
        />
        <div
          class="css-g263yg"
          role="cellContainer-5"
        />
        <div
          class="css-g263yg"
          role="cellContainer-6"
        />
        <div
          class="css-g263yg"
          role="cellContainer-7"
        />
        <div
          class="css-v213q6"
          role="cellContainer-8"
        />
        <div
          class="css-1bqu5zx"
          role="cellContainer-9"
        />
        <div
          class="css-g263yg"
          role="cellContainer-10"
        />
        <div
          class="css-g263yg"
          role="cellContainer-11"
        />
        <div
          class="css-g263yg"
          role="cellContainer-12"
        />
        <div
          class="css-g263yg"
          role="cellContainer-13"
        />
        <div
          class="css-g263yg"
          role="cellContainer-14"
        />
        <div
          class="css-g263yg"
          role="cellContainer-15"
        />
        <div
          style="clear: both;"
        />
      </section>
    </div>
    <div
      class="css-1n2h5h0"
      role="userInfo"
    >
      <div
        class="css-11wxzwa"
        role="grid"
      >
        <p
          class="css-1890bmp"
        >
          Игрок: 
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AppBody Render <AppBody> without data 1`] = `
<DocumentFragment>
  <div
    class="css-cdwrp8"
    role="bodyPanel"
  >
    <div
      class="css-1n2h5h0"
      role="panelHitParade"
    >
      <div
        class="css-11wxzwa"
        role="rootGrid"
      >
        <p
          class="css-1890bmp"
        >
          Хит-парад
        </p>
      </div>
    </div>
    <div
      class="css-1n2h5h0"
    >
      <section
        class="css-rxkl80"
        role="cellsGrid"
      >
        <div
          class="css-1bqu5zx"
          role="cellContainer-0"
        />
        <div
          class="css-g263yg"
          role="cellContainer-1"
        />
        <div
          class="css-g263yg"
          role="cellContainer-2"
        />
        <div
          class="css-g263yg"
          role="cellContainer-3"
        />
        <div
          class="css-g263yg"
          role="cellContainer-4"
        />
        <div
          class="css-g263yg"
          role="cellContainer-5"
        />
        <div
          class="css-g263yg"
          role="cellContainer-6"
        />
        <div
          class="css-g263yg"
          role="cellContainer-7"
        />
        <div
          class="css-v213q6"
          role="cellContainer-8"
        />
        <div
          class="css-1bqu5zx"
          role="cellContainer-9"
        />
        <div
          class="css-g263yg"
          role="cellContainer-10"
        />
        <div
          class="css-g263yg"
          role="cellContainer-11"
        />
        <div
          class="css-g263yg"
          role="cellContainer-12"
        />
        <div
          class="css-g263yg"
          role="cellContainer-13"
        />
        <div
          class="css-g263yg"
          role="cellContainer-14"
        />
        <div
          class="css-g263yg"
          role="cellContainer-15"
        />
        <div
          class="css-g263yg"
          role="cellContainer-16"
        />
        <div
          class="css-v213q6"
          role="cellContainer-17"
        />
        <div
          class="css-1bqu5zx"
          role="cellContainer-18"
        />
        <div
          class="css-g263yg"
          role="cellContainer-19"
        />
        <div
          class="css-g263yg"
          role="cellContainer-20"
        />
        <div
          class="css-g263yg"
          role="cellContainer-21"
        />
        <div
          class="css-g263yg"
          role="cellContainer-22"
        />
        <div
          class="css-g263yg"
          role="cellContainer-23"
        />
        <div
          class="css-g263yg"
          role="cellContainer-24"
        />
        <div
          class="css-g263yg"
          role="cellContainer-25"
        />
        <div
          class="css-v213q6"
          role="cellContainer-26"
        />
        <div
          class="css-1bqu5zx"
          role="cellContainer-27"
        />
        <div
          class="css-g263yg"
          role="cellContainer-28"
        />
        <div
          class="css-g263yg"
          role="cellContainer-29"
        />
        <div
          class="css-g263yg"
          role="cellContainer-30"
        />
        <div
          class="css-g263yg"
          role="cellContainer-31"
        />
        <div
          class="css-g263yg"
          role="cellContainer-32"
        />
        <div
          class="css-g263yg"
          role="cellContainer-33"
        />
        <div
          class="css-g263yg"
          role="cellContainer-34"
        />
        <div
          class="css-v213q6"
          role="cellContainer-35"
        />
        <div
          class="css-1bqu5zx"
          role="cellContainer-36"
        />
        <div
          class="css-g263yg"
          role="cellContainer-37"
        />
        <div
          class="css-g263yg"
          role="cellContainer-38"
        />
        <div
          class="css-g263yg"
          role="cellContainer-39"
        />
        <div
          class="css-g263yg"
          role="cellContainer-40"
        />
        <div
          class="css-g263yg"
          role="cellContainer-41"
        />
        <div
          class="css-g263yg"
          role="cellContainer-42"
        />
        <div
          class="css-g263yg"
          role="cellContainer-43"
        />
        <div
          class="css-v213q6"
          role="cellContainer-44"
        />
        <div
          class="css-1bqu5zx"
          role="cellContainer-45"
        />
        <div
          class="css-g263yg"
          role="cellContainer-46"
        />
        <div
          class="css-g263yg"
          role="cellContainer-47"
        />
        <div
          class="css-g263yg"
          role="cellContainer-48"
        />
        <div
          class="css-g263yg"
          role="cellContainer-49"
        />
        <div
          class="css-g263yg"
          role="cellContainer-50"
        />
        <div
          class="css-g263yg"
          role="cellContainer-51"
        />
        <div
          class="css-g263yg"
          role="cellContainer-52"
        />
        <div
          class="css-v213q6"
          role="cellContainer-53"
        />
        <div
          class="css-1bqu5zx"
          role="cellContainer-54"
        />
        <div
          class="css-g263yg"
          role="cellContainer-55"
        />
        <div
          class="css-g263yg"
          role="cellContainer-56"
        />
        <div
          class="css-g263yg"
          role="cellContainer-57"
        />
        <div
          class="css-g263yg"
          role="cellContainer-58"
        />
        <div
          class="css-g263yg"
          role="cellContainer-59"
        />
        <div
          class="css-g263yg"
          role="cellContainer-60"
        />
        <div
          class="css-g263yg"
          role="cellContainer-61"
        />
        <div
          class="css-v213q6"
          role="cellContainer-62"
        />
        <div
          class="css-1bqu5zx"
          role="cellContainer-63"
        />
        <div
          class="css-g263yg"
          role="cellContainer-64"
        />
        <div
          class="css-g263yg"
          role="cellContainer-65"
        />
        <div
          class="css-g263yg"
          role="cellContainer-66"
        />
        <div
          class="css-g263yg"
          role="cellContainer-67"
        />
        <div
          class="css-g263yg"
          role="cellContainer-68"
        />
        <div
          class="css-g263yg"
          role="cellContainer-69"
        />
        <div
          class="css-g263yg"
          role="cellContainer-70"
        />
        <div
          class="css-v213q6"
          role="cellContainer-71"
        />
        <div
          class="css-1bqu5zx"
          role="cellContainer-72"
        />
        <div
          class="css-g263yg"
          role="cellContainer-73"
        />
        <div
          class="css-g263yg"
          role="cellContainer-74"
        />
        <div
          class="css-g263yg"
          role="cellContainer-75"
        />
        <div
          class="css-g263yg"
          role="cellContainer-76"
        />
        <div
          class="css-g263yg"
          role="cellContainer-77"
        />
        <div
          class="css-g263yg"
          role="cellContainer-78"
        />
        <div
          class="css-g263yg"
          role="cellContainer-79"
        />
        <div
          class="css-v213q6"
          role="cellContainer-80"
        />
        <div
          style="clear: both;"
        />
      </section>
    </div>
    <div
      class="css-1n2h5h0"
      role="userInfo"
    >
      <div
        class="css-11wxzwa"
        role="grid"
      >
        <p
          class="css-1890bmp"
        >
          Игрок: 
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
